.github/workflows/publish.yaml

```
@@ -5,16 +5,44 @@ on:
     types: [created]
 
 jobs:
-  publish-npm:
+  test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - name: Use node version 18
+        uses: actions/setup-node@v1
         with:
-          node-version: 14
-          registry-url: https://registry.npmjs.org/
-      - run: npm i
-      - run: node test/validateModuleExportsMatchCommonJS/index.js
-      - run: npm publish
-        env:
-          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
+          node-version: 18
+      - name: Update npm
+        run: npm i -g npm@8.19.4
+      - name: Run tests
+        run: npm run test
+  build:
+    needs: [test]
+    permissions:
+      id-token: write # For signing
+      contents: read # For repo checkout.
+      actions: read # For getting workflow run info.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_nodejs_slsa3.yml@v1.7.0

```

v1.8.0 fixes some issues with unscoped packages.
```suggestion
    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_nodejs_slsa3.yml@v1.8.0
```